estate/models/estate_property.py

```diff
@@ -78,3 +78,24 @@ def _onchange_garden(self):
             self.garden_area = 0
             self.garden_orientation = False
 
+
+    def action_sold(self):
+        for record in self:
+            if record.state == "canceled":
+                raise exceptions.UserError("Canceled property cannot be sold")
+            elif record.state == "sold":
+                raise exceptions.UserError("Property is already sold")
```

Comment:
This is OK if we need specific error messages

another alternative could be this:

```py
if record.state in ["canceled", "sold"]:
    raise exceptions.UserError("A property with state %s cannot be changed to sold." % record.state)
```